### PR TITLE
ENG-1644: Extend web Traktor app to work with encrypted backups

### PR DESCRIPTION
**Pull request by workaandrey**

### CODE REVIEW

File: components/wpress-browser.js [hunk 1]
```diff
@@ -208,14 +237,35 @@ export default class WPressBrowser extends React.Component {
...
+            crypto.decryptFile(this.state.decryptionKey, file.content)
+                .then(fileContent => {
+                    const blob = new Blob([fileContent]);
+
+                    saveAs(blob, file.name);
+
+                    self.setState({loading: false});
+                });
+        } else {
+            saveAs('file.content', file.name)
```
Comment (yani-) [inline components/wpress-browser.js]: I think your editor wrongly quoted this variable :)

File: components/wpress-browser.js [hunk 2]
```diff
@@ -2,7 +2,10 @@ import path from 'path'
 import React from 'react'
 import cn from 'classnames'
 import { saveAs } from 'file-saver'
+import {Modal} from "./Modal";
```
Comment (yani-) [inline components/wpress-browser.js]: `{ Modal }` makes it slightly more readable

Comment (borislav-angelov) [reply components/wpress-browser.js]: @yani- can you fix this as well as the two lines below

File: components/wpress-browser.js [hunk 3]
```diff
@@ -129,6 +146,23 @@ export default class WPressBrowser extends React.Component {
         this.readFile(file)
     }
 
+    readPackageJson(file, size) {
+        let reader = new FileReader();
+        const self = this;
+        reader.addEventListener("loadend", () => {
+            const content = (new TextDecoder('utf-8')).decode(new DataView(reader.result));
+            const config = JSON.parse(content);
+            if(config.Encrypted) {
```
Comment (borislav-angelov) [inline components/wpress-browser.js]: add space after if and before (

File: components/wpress-browser.js [hunk 4]
```diff
@@ -153,6 +187,9 @@ export default class WPressBrowser extends React.Component {
                     node = !!foundNode ? foundNode : node.addChild(new Node(parent))
                 })
             }
+            if(name == 'package.json') {
```
Comment (borislav-angelov) [inline components/wpress-browser.js]: add space after if and before ( and add triple ===

File: components/wpress-browser.js [hunk 5]
```diff
@@ -208,14 +245,42 @@ export default class WPressBrowser extends React.Component {
     onFileClick(file, event) {
         event.preventDefault()
         event.stopPropagation()
-        saveAs(file.content, file.name)
+
+        if(this.state.loading) {
```
Comment (borislav-angelov) [inline components/wpress-browser.js]: add space after if and before (

File: components/wpress-browser.js [hunk 6]
```diff
@@ -208,14 +245,42 @@ export default class WPressBrowser extends React.Component {
...
+            return false;
+        }
+
+        this.setState({loading: file.name});
+        const self = this;
+
+        if(this.state.decryptionKey) {
```
Comment (borislav-angelov) [inline components/wpress-browser.js]: add space after if and before (

File: components/wpress-browser.js [hunk 7]
```diff
@@ -252,14 +318,82 @@ export default class WPressBrowser extends React.Component {
...
+            console.error(e);
+
+            this.setState({
+                passwordError : 'Invalid password',
+            })
+        }
+    }
+
     render () {
+        if(this.state.isPasswordRequested) {
```
Comment (borislav-angelov) [inline components/wpress-browser.js]: add space after if and before (

File: helpers/cryptoHelpers.js [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+
+export async function decryptFile(decryptionKey, fileContent) {
+    const arrayBuffer = await fileContent.arrayBuffer();
+    let buffer = Buffer.from(arrayBuffer);
+
+    let decryptedParts = Buffer.from([]);
+    let bufferLength = buffer.length;
+
+    const chunksNumber = Math.ceil(bufferLength / CHUNK_LENGTH);
+    for(let i = 0; i < chunksNumber; i++ ) {
```
Comment (borislav-angelov) [inline helpers/cryptoHelpers.js:42]: add space after for and before (

**Pull request merged**